### PR TITLE
Cypress: Set up frontend image snapshot testing

### PR DESCRIPTION
**Pull request by topocount**
These changes install a dom/image snapshot testing plugin for Cypress
and initialize snapshots and snapshot tests for the frontend.

No CI integration is configured here. This just sets up the tests within
cypress itself, so environmental configs are out of scope at the moment.

**nb** I realize that this might be under-documented, and I'm very open to suggestions
on how to best share details behind some of these configs.

test plan: `yarn start > /dev/null & yarn cypress run snapshot --headless --browser chrome`

### CODE REVIEW

**Review by BrianBelhumeur — APPROVED** [review]
Snapshots can be a very useful tool when there are no good alternatives, but I encourage sparse use of them due to their inherent brittleness (e.g. a single minor CSS change can break every snapshot test).

Having a baseline "is the admin UI even loading" is a good start, and we may eventually replace the snapshots with more robust Cypress tests.

File: cypress/integration/snapshot.spec.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    // this step ensures the frontend has loaded
+    // before taking a snapshot
+    cy.get('[href="#/explorer"]');
+    // $FlowIgnore[prop-missing]
+    cy.document().toMatchImageSnapshot();
+  });
+}
+
+context("Snapshots", () => {
+  it("renders the Explorer view", () => {
```
Comment (decentralion) [inline cypress/integration/snapshot.spec.js:33]: I notice these are going to map onto file names directly (e.g. `/integration/__image_snapshots__/Snapshots renders the Explorer view #0.png`). I usually see capitalization and spaces in the name like this as a code smell. Is this the cypress default? Is it configurable?

File: cypress/integration/snapshot.spec.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+function getSnapShotOnLoad(route: string) {
+  cy.visit(route).then(() => {
+    // this step ensures the frontend has loaded
+    // before taking a snapshot
+    cy.get('[href="#/explorer"]');
+    // $FlowIgnore[prop-missing]
+    cy.document().toMatchImageSnapshot();
+  });
+}
```
Comment (decentralion) [inline cypress/integration/snapshot.spec.js:30]: From a performance standpoint: we load the page from scratch every time (including waiting for loading the cred data). Is there a way we can load the site once and then navigate around it?

**Review by decentralion — COMMENTED** [review]
so... what's up with all the lint errors?

File: cypress/integration/snapshot.spec.js (hunk 2 above)
Comment (topocount) [reply cypress/integration/snapshot.spec.js:30]: yes we can definitely do that. The test won't self-document quite as nicely but i think it's worth it. I was actually anticipating this suggestion

**Comment by topocount** [comment]
> so... what's up with all the lint errors?

The solution to fixing them was deleting and rebuilding yarn.lock so I'm not really sure.

**Comment by topocount** [comment]
Will reopen this and #2436 to get them merged later

**Pull request closed without merging**